### PR TITLE
Not applying network config during session restart

### LINKED ISSUE #1886: Network config should not be applied when a session process exits on the VCH
Currently a "reload" is triggered when a process on the VCH crashes, which in turn reconfigures the network. This should not happen if it succeeded the first time around. Only the process launch should be attempted again.

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #1886

### CODE REVIEW

**Comment by vburenin** [comment]
LGTM

**Comment by jzt** [comment]
LGTM

File: lib/tether/ops_linux.go [hunk 1]
```diff
@@ -425,23 +425,34 @@ func (t *BaseOperations) Apply(endpoint *NetworkEndpoint) error {
...
 	// Locate interface
 	slot, err := strconv.Atoi(endpoint.ID)
 	if err != nil {
-		detail := fmt.Sprintf("endpoint ID must be a base10 numeric pci slot identifier: %s", err)
-		return errors.New(detail)
+		return fmt.Errorf("endpoint ID must be a base10 numeric pci slot identifier: %s", err)
 	}
-	link, err := nl.LinkBySlot(int32(slot))
+
+	defer func() {
```
Comment (vburenin) [inline lib/tether/ops_linux.go]: I really dislike this pattern when deferred function checks for errors and set some flags, it should be done by a higher level caller. I see benefits of this approach, I understand the reason, it just makes this code so hard to follow. I needed to unroll so much code to ensure I am not missing anything. This function has to be already split into multiple smaller chunks to make it simpler.

Comment (hmahmood) [inline lib/tether/ops_linux.go]: Hm... actually I think we return early in at least one case below. For now I am going to leave this alone.
